code/controllers/subsystem/who.dm

```diff
@@ -0,0 +1,325 @@
+SUBSYSTEM_DEF(who)
+	name = "Who"
+	flags = SS_NO_INIT|SS_BACKGROUND
+	runlevels = RUNLEVELS_DEFAULT|RUNLEVEL_LOBBY
+	wait = 5 SECONDS //smooth update
+
+	//make it global, good for CPU, no per player UI update
+	var/datum/player_list/who = new
+	var/datum/player_list/staff/staff_who = new
+
+/datum/controller/subsystem/who/fire(resumed = TRUE)
+	who.update_data()
+	staff_who.update_data()
+
+//datum
+/datum/player_list
+	var/tgui_name = "Who"
+	var/tgui_interface_name = "Who"
+	var/list/mobs_ckey = list()
+	var/list/list_data = list()
+
+/datum/player_list/proc/update_data()
+	var/list/new_list_data = list()
+	var/list/new_mobs_ckey = list()
+	var/list/additiona_data = list(
+		"lobby" = 0,
+		"admin_observers" = 0,
+		"observers" = 0,
+		"yautja" = 0,
+		"infected_preds" = 0,
+		"humans" = 0,
+		"infected_humans" = 0,
+		"uscm" = 0,
+		"uscm_marines" = 0,
+	)
+	var/list/mobs_by_faction = list()
+	new_list_data["additional_info"] = list()
+
+	for(var/client/client in GLOB.clients)
+		new_list_data["all_clients"]++
+		var/list/client_payload = list()
+		client_payload["ckey"] = "[client.key]"
+		client_payload["text"] = "[client.key]"
+		client_payload["ckey_color"] = "white"// change it if want! for byond supporters maybe? (client.player_data?.donator_info.patreon_function_available("ooc_color") ? "#D4AF37" : "white")
+		var/mob/client_mob = client.mob
+		new_mobs_ckey[client.key] = client_mob
+		if(client_mob)
+			if(istype(client_mob, /mob/new_player))
+				client_payload["text"] += " - in Lobby"
```

Comment:
I'm not sure if we want to directly change the display code in the backend. I feel that this couples the backend code with the render code quite tightly. I believe that a looser approach of say, a flag variable of being in lobby, would be preferable.